File: java/design-patterns/facade/extend/Extend.java
class HomeTheaterFacade {
    void watchMovie(String title) { /* …same one call for callers… */ }

    void endMovie() {            // a brand-new high-level task
        box.stop();
        projector.off();
        amp.off();
        lights.dim(100);
    }
}

// Callers gain a whole feature without learning any subsystem:
theater.endMovie();

File: typescript/design-patterns/facade/extend/extend.ts
class HomeTheaterFacade {
  watchMovie(title: string) { /* …same one call for callers… */ }

  endMovie() {                 // a brand-new high-level task
    this.box.stop();
    this.projector.off();
    this.amp.off();
    this.lights.dim(100);
  }
}

// Callers gain a whole feature without learning any subsystem:
theater.endMovie();
// Swap StreamingBox for a new vendor, or re-order the steps —
// every caller's code stays exactly as it was.